bmx280.c

```diff
@@ -309,22 +405,55 @@ static esp_err_t bmx280_calibrate(bmx280_t *bmx280)
     return ESP_OK;
 }
 
-bmx280_t* bmx280_create(i2c_port_t port)
+#if !(CONFIG_USE_I2C_MASTER_DRIVER)
+bmx280_t* bmx280_create_legacy(i2c_port_t port)
 {
     bmx280_t* bmx280 = malloc(sizeof(bmx280_t));
     if (bmx280)
     {
         memset(bmx280, 0, sizeof(bmx280_t));
-
         bmx280->i2c_port = port;
         bmx280->slave = 0xDE;
         bmx280->chip_id = 0xAD;
     }
+    else
+    {
+        ESP_LOGE("bmx280", "Failed to allocate memory for bmx280.");
+        bmx280_close(bmx280);
+        return NULL;
+    }
     return bmx280;
 }
+#else
+bmx280_t* bmx280_create_master(i2c_master_bus_handle_t bus_handle)
+{
+    bmx280_t* bmx280 = malloc(sizeof(bmx280_t));
+    if (bmx280)
+    {
+        memset(bmx280, 0, sizeof(bmx280_t));
+        bmx280->bus_handle = bus_handle;
+        bmx280->dev_cfg.dev_addr_length = I2C_ADDR_BIT_LEN_7;
+        bmx280->dev_cfg.device_address = 0xDE;
+        bmx280->dev_cfg.scl_speed_hz =CONFIG_BMX280_I2C_CLK_SPEED_HZ;
+        bmx280->i2c_dev = NULL;
+        bmx280->chip_id = 0xAD;
+    }
+    else
+    {
+        ESP_LOGE("bmx280", "Failed to allocate memory for bmx280.");
+        bmx280_close(bmx280);
+        return NULL;
+    }
+    return bmx280;
+}
+#endif
 
 void bmx280_close(bmx280_t *bmx280)
 {
+    #if CONFIG_USE_I2C_MASTER_DRIVER
+    if(bmx280->i2c_dev!=NULL)
```

Comment:
This still dereferences a pointer that may not be initialized (The `bmx280` driver structure as well).

```c
void bmx280_close(bmx280_t *bmx280)
{
    #if CONFIG_USE_I2C_MASTER_DRIVER
    if (bmx280 != NULL && bmx280->i2c_dev != NULL)
    {
        i2c_master_bus_rm_device(bmx280->i2c_dev);
    }
    #endif
    free(bmx280);
}
```

Notice I also check for the pointer bmx280 itself.